tests/functional/test_scoping/test_rrel.py

```diff
@@ -0,0 +1,284 @@
+from __future__ import unicode_literals
+from textx.scoping.rrel import rrel_standalone, parse
+from arpeggio import ParserPython
+from textx import metamodel_from_str
+from textx.scoping.rrel import find
+
+
+def test_rrel_basic_parser1():
+    parser = ParserPython(rrel_standalone)
+    parse_tree = parser.parse("^pkg*.cls")
+    assert len(parse_tree) == 2  # always true (one path, one EOF)
+
+    parse_tree = parser.parse("obj.ref.~extension *.methods")
+    assert len(parse_tree) == 2  # always true (one path, one EOF)
+
+    parse_tree = parser.parse("instance.(type.vals)*")
+    assert len(parse_tree) == 2  # always true (one path, one EOF)
+
+
+def test_rrel_basic_parser2():
+    tree = parse("^pkg*.cls")
+    assert str(tree) == '(..)*.(pkg)*.cls'
+    tree = parse("obj.ref.~extension *.methods")
+    assert str(tree) == 'obj.ref.(~extension)*.methods'
+    tree = parse("type.vals")
+    assert str(tree) == 'type.vals'
+    tree = parse("(type.vals)")
+    assert str(tree) == '(type.vals)'
+    tree = parse("(type.vals)*")
+    assert str(tree) == '(type.vals)*'
+    tree = parse("instance . ( type.vals ) *")
+    assert str(tree) == 'instance.(type.vals)*'
+    tree = parse("a,b,c")
+    assert str(tree) == 'a,b,c'
+
+
+metamodel_str = '''
+    Model:
+        packages*=Package
+    ;
+
+    Package:
+        'package' name=ID '{'
+        packages*=Package
+        classes*=Class
+        '}'
+    ;
+
+    Class:
+        'class' name=ID '{'
+            attributes*=Attribute
+        '}'
+    ;
+
+    Attribute:
+            'attr' name=ID ';'
+    ;
+
+    Comment: /#.*/;
+    FQN: ID('.'ID)*;
+    '''
+
+modeltext = '''
+    package P1 {
+        class Part1 {
+        }
+    }
+    package P2 {
+        package Inner {
+            class Inner {
+                attr inner;
+            }
+        }
+        class Part2 {
+            attr rec;
+        }
+        class C2 {
+            attr p1;
+            attr p2a;
+            attr p2b;
+        }
+        class rec {
+            attr p1;
+        }
+    }
+    '''
+
+
+def test_rrel_basic_lookup():
+    """
+    This is a basic test for the find function:
+    we use a model with some structure
+    and query this structure with RREL expressions.
+    """
+    #################################
+    # META MODEL DEF
+    #################################
+
+    my_metamodel = metamodel_from_str(metamodel_str)
+
+    #################################
+    # MODEL PARSING
+    #################################
+
+    my_model = my_metamodel.model_from_str(modeltext)
+
+    #################################
+    # TEST
+    #################################
+
+    P2 = find(my_model, "P2", "packages")
+    assert P2 is not None
+    Part2 = find(my_model, "P2.Part2", "packages.classes")
+    assert Part2 is not None
+    rec = find(my_model, "P2.Part2.rec", "packages.classes.attributes")
+    assert rec is not None
```

Comment:
Maybe it would be better to assert the name or the type of the found object in all these asserts (`assert rec.name == 'rec'`). It would be a stronger assert.

Reply:
Changed the test: now, we test the name instead of "not None". Also removed redundant tests for "not None", if we access an attribute in an assert afterwards... Added some more checks to make the tests even stronger.